code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -1,339 +1,298 @@
+		timeleft--
+		if(timeleft <= 0)
+			explode()
 			return
+		updateUsrDialog()
 
-		auth = I
-
-	if(!anchored)
-		return
-
-	switch(removal_stage)
-		if(0)
-			if(!iswelder(I))
-				return
 
-			var/obj/item/tool/weldingtool/WT = I
-			if(!WT.isOn()) 
-				return
-
-			if(WT.get_fuel() < 5) // uses up 5 fuel.
-				to_chat(user, "<span class='warning'>You need more fuel to complete this task.</span>")
-				return
+/obj/machinery/nuclearbomb/start_processing()
+	. = ..()
+	countdown.start()
 
-			user.visible_message("<span class='notice'>[user] starts cutting loose the anchoring bolt covers on [src].</span>",
-			"<span class='notice'>You start cutting loose the anchoring bolt covers with [I].</span>")
-			if(!do_after(user, 40, TRUE, src, BUSY_ICON_BUILD, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)) || !WT.remove_fuel(5, user))
-				return
 
-			user.visible_message("<span class='notice'>[user] cuts through the bolt covers on [src].</span>",
-			"<span class='notice'>You cut through the bolt cover.</span>")
-			removal_stage = 1
+/obj/machinery/nuclearbomb/stop_processing()
+	countdown.stop()
+	return ..()
 
-		if(1)
-			if(!iscrowbar(I))
-				return
 
-			user.visible_message("<span class='notice'>[user] starts forcing open the bolt covers on [src].</span>",
-			"<span class='notice'>You start forcing open the anchoring bolt covers with [I].</span>")
-			
-			if(!do_after(user, 15, TRUE, src, BUSY_ICON_BUILD))
-				return
+/obj/machinery/nuclearbomb/proc/explode()
+	stop_processing()
 
-			user.visible_message("<span class='notice'>[user] forces open the bolt covers on [src].</span>",
-			"<span class='notice'>You force open the bolt covers.</span>")
-			removal_stage = 2
+	if(safety)
+		timer_enabled = FALSE
+		GLOB.nukes_set_list -= src
+		return
 
-		if(2)
-			if(!iswelder(I))
-				return
+	if(exploded)
+		return
+	exploded = TRUE
+	safety = TRUE
+	if(!lighthack)
+		icon_state = "nuclearbomb3"
+
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_NUKE_EXPLODED)
+	addtimer(CALLBACK(src, .proc/BurnEveryoneOnZLevel, z), 5 SECONDS)
+	play_cinematic() //The round ends as soon as this happens, or it should.
+	return TRUE
 
-			var/obj/item/tool/weldingtool/WT = I
 
-			user.visible_message("<span class='notice'>[user] starts cutting apart the anchoring system sealant on [src].</span>",
-			"<span class='notice'>You start cutting apart the anchoring system's sealant with [I].</span>")
+/obj/machinery/nuclearbomb/proc/play_cinematic()
+	GLOB.enter_allowed = FALSE
+	priority_announce("DANGER. DANGER. Planetary Nuke Activated. DANGER. DANGER. Self destruct in progress. DANGER. DANGER.", "Priority Alert")
+	SEND_SOUND(world, pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg'))
 
-			if(!do_after(user, 40, TRUE, src, BUSY_ICON_BUILD, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)) || !WT.remove_fuel(5, user))
-				return
+	for(var/x in GLOB.player_list)
+		var/mob/M = x
+		if(isobserver(M))
```

Comment:
This will still happen for new_player

Reply:
That's how the SD explosion works. Is that a problem?
The whole proc should be discussed, though, as it's a copypaste from `initiate_self_destruct()`
As it is, the shaking won't be seen. It will be covered by the cinematic screen.
At some point a rework may have skipped a `spawn()` or something for that to happen.

Reply:
Not really a problem, but an elegant solution may be to define the proc to just return at the mob level, and to shake at the living level, this would save us both typechecks

Reply:
The name of that is a signal, really. I'll see later if there's a simple way of implementing it.